Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py

```diff
@@ -4646,6 +4646,98 @@ def test_invalid_b_type_init(self):
             assert "Invalid entity B type: DomainTest" in str(exception)
 
 
+class TestSetAndGetLastRun:
+
+    def test_get_last_run_in_6_2_when_get_last_run_has_results(self, mocker):
+        """
+        Given: 6.2.0 environment
+        When: Fetch indicators when getLastRun returns results
+        Then: Returning all indicators from demisto.getLastRun object
+        """
+        import demistomock as demisto
+        from CommonServerPython import get_feed_last_run
+        mocker.patch('CommonServerPython.get_demisto_version', return_value={"version": "6.2.0"})
+        mocker.patch.object(demisto, 'getLastRun', return_value={1: "first indicator"})
+        result = get_feed_last_run()
+        assert result == {1: "first indicator"}
+
+    def test_get_last_run_in_6_1_when_get_integration_context_has_results(self, mocker):
+        """
+        Given: 6.1.0 environment
+        When: Fetch indicators when getIntegrationContext return results:
+                This can happen when updating XSOAR version to 6.2.0 while a feed instance is already set.
+        Then: Returning all indicators from demisto.getIntegrationContext object
+        """
+        import demistomock as demisto
+        from CommonServerPython import get_feed_last_run
+        mocker.patch('CommonServerPython.get_demisto_version', return_value={"version": "6.1.0"})
+        mocker.patch.object(demisto, 'getIntegrationContext', return_value={1: "first indicator"})
+        result = get_feed_last_run()
+        assert result == {1: "first indicator"}
+
+    def test_get_last_run_in_6_2_when_get_last_run_has_no_results(self, mocker):
+        """
+        Given: 6.2.0 environment
+        When: Fetch indicators when getLastRun and getIntegrationContext are empty
+        Then: {}
+        """
+        import demistomock as demisto
+        from CommonServerPython import get_feed_last_run
+        mocker.patch('CommonServerPython.get_demisto_version', return_value={"version": "6.2.0"})
+        mocker.patch.object(demisto, 'getIntegrationContext', return_value={})
+        mocker.patch.object(demisto, 'getLastRun', return_value={})
+        result = get_feed_last_run()
+        assert result == {}
+
+    def test_get_last_run_in_6_2_when_get_last_is_empty_and_get_integration_is_not(self, mocker):
+        """
+        Given: 6.2.0 environment
+        When: Fetch indicators when getLastRun is empty and getIntegrationContext has results.
```

Comment:
"setLastRun is empty and getIntegrationContext has results." is part of the "Given"

Reply:
done